rest/model/dns/zone.go

```diff
@@ -51,6 +53,46 @@ type Zone struct {
 	Tags map[string]string `json:"tags,omitempty"` // Only relevant for DDI
 }
 
+func (z Zone) MarshalJSON() ([]byte, error) {
+	type Alias Zone
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(&z),
+	}
+
+	if z.NetworkIDs != nil {
+		aux.Networks = &z.NetworkIDs
+	}
+
+	if aux.Networks == nil {
+		aux.Networks = &[]int{}
+	}
```

Comment:
@arnavbhattacharya-NS1 can you do this?
```suggestion
} else {
	aux.Networks = nil
}
```